File: articles/causal-algorithms-and-intervention-modeling/calculators/python/intervention_effect_calculator.py
from __future__ import annotations
import argparse


def main() -> None:
    parser = argparse.ArgumentParser(description="Compute a simple intervention effect contrast.")
    parser.add_argument("--baseline-outcome", type=float, required=True)
    parser.add_argument("--intervention-outcome", type=float, required=True)
    args = parser.parse_args()
    effect = args.intervention_outcome - args.baseline_outcome
    print(f"baseline_outcome={args.baseline_outcome:.6f}")
    print(f"intervention_outcome={args.intervention_outcome:.6f}")
    print(f"estimated_effect={effect:.6f}")


if __name__ == "__main__":
    main()

File: articles/causal-algorithms-and-intervention-modeling/calculators/r/intervention_effect_calculator.R
args <- commandArgs(trailingOnly = TRUE)
if (length(args) < 2) stop("Usage: Rscript intervention_effect_calculator.R <baseline_outcome> <intervention_outcome>")
baseline <- as.numeric(args[1])
intervention <- as.numeric(args[2])
effect <- intervention - baseline
cat(sprintf("baseline_outcome=%.6f\n", baseline))
cat(sprintf("intervention_outcome=%.6f\n", intervention))
cat(sprintf("estimated_effect=%.6f\n", effect))
